src/bsoncxx/json.hpp

```diff
@@ -34,18 +34,39 @@ BSONCXX_INLINE_NAMESPACE_BEGIN
 extern const bool k_silence_doxygen;
 
 ///
-/// Converts a BSON document to a JSON string.
+/// An enumeration of the types of Extended JSON that the to_extended_json function accepts
+///
+/// The values are:
+///   k_canonical: to produce Canonical Extended JSON
+///   k_relaxed: to produce Relaxed Extended JSON
+enum class ExtendedJsonMode { k_canonical, k_relaxed };
```

Comment:
IIRC, it was guidance for public enums that they always declare a size class so that adding new enumerators with explicit values could never change the size leading to an ABI break. It appears that that rule has not been rigorously followed.

Reply:
I'm happy to add it, but as you also said before (paraphrased) "we don't handle ABI breaks".

Reply:
This entire project has been undertaken in an effort to set an ABI. The fact that we haven't managed to do it yet isn't an argument against doing the engineering right so that we *can* do it in the future, and have a shot at maintaining an ABI once we do.